src/reality-capture.ts

```diff
@@ -0,0 +1,273 @@
+}
+
+export interface IPhotoSceneOutput {
+    Photoscene: {
+        filesize: string;
+        photosceneid: string;
+        progress: string;
+        progressmsg: string;
+        resultmsg: string;
+        scenelink: string;
+        urn: string;
+    }
+}
+
+export interface IPhotoSceneProcess {
+    msg: string;
+    Photoscene: {
+        photosceneid: string;
+    }
+}
+
+export interface IPhotoSceneProgress {
+    photosceneid: string;
+    progressmsg: string;
+    progress: string;
+}
+
+/**
+ * Client providing access to Autodesk Forge {@link https://forge.autodesk.com/en/docs/reality-capture/v1/developers_guide/overview|reality capture APIs}.
+ * @tutorial realitycapture
+ */
+export class RealityCaptureClient extends ForgeClient {
+    /**
+     * Initializes new client with specific authentication method
+     * @param {IAuthOptions} auth Authentication object,
+     * containing `client_id` and `client_secret` properties (for 2-legged authentication).
+     * @param {string} [host="https://developer.api.autodesk.com"] Forge API host.
+     */
+    constructor(auth: IAuthOptions, host?: string) {
+        super('photo-to-3d/v1/', auth, host);
+    }
+
+    /**
+     * Creates new photoscene
+     * {@link https://forge.autodesk.com/en/docs/reality-capture/v1/reference/http/photoscene-POST|docs}.
+     * @async
+     * @param {string} scenename Specifies the name of the photoscene.
+     * @param {string} callback URI to be invoked when the processing of the photoscene has completed.
+     * @param {OutputFormat} format Output file format. Multiple file formats can be listed in a comma-delimited list.
+     * @param {SceneType} scenetype Specifies the subject type of the photoscene.
+     * @param {GpsType} gpstype Specifies the GPS coordinates type.
+     * @param {string} hubprojectid The identifier of an A360 Personal hub or BIM 360 Docs project. Output files will be delivered to a folder in this project.
+     * @param {string} hubfolderid The URN of an A360 Personal hub or BIM 360 Docs folder. Output files will be delivered to this folder.
+     * @param {EngineVersion} version The reconstruction engine version.
+     * @param {string[]} metadataname Fine-tuning parameter field names indexed to match the metadata_value index, beginning from index 0.
+     * @param {string[]} metadatavalue Fine-tuning parameter values indexed to match the metadata_name index, beginning from index 0.
+     * @returns {Promise<IPhotoScene>} A JSON object containing details of the photoscene that was created, with property 'photosceneid' ID of the photoscene that was created.
+     * @throws Error when the request fails, for example, due to invalid request.
+     */
+    async createPhotoScene(scenename: string, callback: string, format: OutputFormat, scenetype: SceneType, gpstype?: GpsType, hubprojectid?: string, hubfolderid?: string, version?: EngineVersion, metadataname?: string[], metadatavalue?: string[]): Promise<IPhotoScene> {
```

Comment:
I'm wondering if we should perhaps group all the optional arguments into an "options" object, what do you think? That way, if I just want to define the engine version, I could do something like `await client.createPhotoScene('scenename', 'callback', format, scenetype, { version: XYZ });`.

Are all the other function arguments (scenename, callback, format, scenetype) always required?

Reply:
only scenename is required. format will default to rcm. scene type defaults to aerial.

Reply:
Fixed in https://github.com/mazerab/forge-server-utils/commit/b247de34cee8083cb595df5f81106f09a0a92366